src/borg/archiver.py

```diff
@@ -442,17 +439,13 @@ def _process(self, archive, cache, matcher, exclude_caches, exclude_if_present,
                 if not dry_run:
                     status = archive.process_dir(path, st)
                 if recurse:
-                    try:
+                    with backup_io('scandir'):
                         entries = helpers.scandir_inorder(path)
-                    except OSError as e:
-                        status = 'E'
-                        self.print_warning('%s: scandir: %s', path, e)
-                    else:
-                        for dirent in entries:
```

Comment:
this was an except/else so that it could reach the print_file_status below. Since the exception try-scope ends before that, this is not necessary anymore.